Index out of bounds exception with HTTP-Kafka proxy

**Describe the bug**
When I execute the GET request with a `body`, I get the index out-of-bounds error. While sending data on a GET is out of spec, it isn't ideal that the server stops.


Apis are being accessed from the [Postman REST template](https://go.postman.co/redirect/workspace?type=personal&collectionTemplateId=ddb19591-3097-41cf-82af-c84273e56719) 

simple GET returns 200:
```
curl -I 'http://localhost:8080/sse/events'
```

adding the header returns 200:
```
curl -I --location --request GET 'http://localhost:8080/sse/events' --header 'Content-Type: application/json'
```

GET with a json body results in an error:
```
curl --location --request GET 'http://localhost:8080/sse/events' \
--header 'Content-Type: application/json' \
--data '{
	"name": "Add your name in the body"
}'
```

Error:
```
started
org.agrona.concurrent.AgentTerminationException: java.lang.IndexOutOfBoundsException: index=-328 length=368 capacity=65536
        at io.aklivity.zilla.runtime.engine@0.9.49/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:645)
        at org.agrona.core/org.agrona.concurrent.AgentRunner.doDutyCycle(AgentRunner.java:291)
        at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:164)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IndexOutOfBoundsException: index=-328 length=368 capacity=65536
        at org.agrona.core/org.agrona.concurrent.UnsafeBuffer.boundsCheck0(UnsafeBuffer.java:1692)
        at org.agrona.core/org.agrona.concurrent.UnsafeBuffer.boundsCheck(UnsafeBuffer.java:1698)
        at org.agrona.core/org.agrona.concurrent.UnsafeBuffer.putBytes(UnsafeBuffer.java:946)
        at io.aklivity.zilla.runtime.binding.http@0.9.49/io.aklivity.zilla.runtime.binding.http.internal.stream.HttpServerFactory$HttpServer.decodeNetwork(HttpServerFactory.java:2109)
        at io.aklivity.zilla.runtime.binding.http@0.9.49/io.aklivity.zilla.runtime.binding.http.internal.stream.HttpServerFactory$HttpServer.decodeNetworkIfBuffered(HttpServerFactory.java:2074)
        at io.aklivity.zilla.runtime.binding.http@0.9.49/io.aklivity.zilla.runtime.binding.http.internal.stream.HttpServerFactory$HttpServer$HttpExchange.onRequestWindow(HttpServerFactory.java:2896)
        at io.aklivity.zilla.runtime.binding.http@0.9.49/io.aklivity.zilla.runtime.binding.http.internal.stream.HttpServerFactory$HttpServer$HttpExchange.onExchange(HttpServerFactory.java:2827)
        at io.aklivity.zilla.runtime.engine@0.9.49/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleReadInitial(DispatchAgent.java:1017)
        at io.aklivity.zilla.runtime.engine@0.9.49/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleRead(DispatchAgent.java:952)
        at io.aklivity.zilla.runtime.engine@0.9.49/io.aklivity.zilla.runtime.engine.internal.concurent.ManyToOneRingBuffer.read(ManyToOneRingBuffer.java:181)
        at io.aklivity.zilla.runtime.engine@0.9.49/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:639)
        ... 3 more
        Suppressed: java.lang.Exception: [engine/data#3]        [0x0303000000000005] streams=[consumeAt=0x000005c8 (0x00000000000005c8), produceAt=0x000006c0 (0x00000000000006c0)]
                at io.aklivity.zilla.runtime.engine@0.9.49/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:643)
                ... 3 more
stopped
```


**To Reproduce**

zilla.yaml
```yaml
name: quickstart
bindings:

# Gateway ingress config
  api_tcp_server:
    type: tcp
    kind: server
    options:
      host: 0.0.0.0
      port: 8080
    exit: http_server
  http_server:
    type: http
    kind: server
    options:
      access-control:
        policy: cross-origin
    routes:
      - when:
          - headers:
              :scheme: http
              :authority: localhost:8080
              content-type: application/json
          - headers:
              :scheme: http
              :authority: localhost:8080
        exit: http_kafka_proxy

# Proxy REST endpoints to Kafka a topic
  http_kafka_proxy:
    type: http-kafka
    kind: proxy
    routes:
      - when:
          - method: POST
            path: /sse/events
        exit: kafka_cache_client
        with:
          capability: produce
          topic: sse-events
          key: ${idempotencyKey}
      - when:
          - method: GET
            path: /sse/events
        exit: kafka_cache_client
        with:
          capability: fetch
          topic: sse-events
          merge:
            content-type: application/json

# Kafka caching layer
  kafka_cache_client:
    type: kafka
    kind: cache_client
    options:
      bootstrap:
        - sse-events
    exit: kafka_cache_server
  kafka_cache_server:
    type: kafka
    kind: cache_server
    exit: kafka_client

# Connect to local Kafka
  kafka_client:
    type: kafka
    kind: client
    exit: kafka_tcp_client
  kafka_tcp_client:
    type: tcp
    kind: client
    options:
      host: kafka
      port: 29092
    routes:
      - when:
          - cidr: 0.0.0.0/0
```

**Expected behavior**
Return a 400, ignore the payload, or handle it for passthrough situations where someone created a weird workflow with JSON data on GET. 

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: MACOS, docker

--- a/runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpServerFactory.java
+++ b/runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpServerFactory.java
@@ -2732,7 +2732,7 @@ public final class HttpServerFactory implements HttpStreamFactory
                 Flyweight extension)
             {
                 int requestNoAck = (int)(requestSeq - requestAck);
-                int length = Math.min(requestMax - requestNoAck - requestPad, limit - offset);
+                int length = Math.min(Math.max(requestMax - requestNoAck - requestPad, 0), limit - offset);
 
                 if (length > 0)
                 {

